### PR TITLE
JVM Working Directory Feature

### PR DESCRIPTION
**Pull request by hlhoots**
#build

### CODE REVIEW

**Comment by hlhoots** [comment]
#build

**Comment by hlhoots** [comment]
rebase and rebuild with minor updates.

#build

**Comment by hlhoots** [comment]
#build

**Comment by hlhoots** [comment]
This PR is for Issue #15128

**Comment by hlhoots** [comment]
rebase and 
#build

**Comment by hlhoots** [comment]
Last build had 1 ui failure that was unrelated to these changes.

**Comment by hlhoots** [comment]
Missing relative path changes in last build.  Added and rebuilding.
#build

**Comment by hlhoots** [comment]
#build

**Comment by hlhoots** [comment]
#build

**Comment by hlhoots** [comment]
javacore was generated in /servers/workingServerDir which the test frame work didnt like.  moving to a parent location higher up which should fix the issue.

#build

**Comment by hlhoots** [comment]
good build above.

**Comment by hlhoots** [comment]
windows build: https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_DD1MkDyvEeygsrM6ShizkA

**Comment by hlhoots** [comment]
https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_Vyim8D46EeygsrM6ShizkA&tab=com.ibm.team.build.web.ui.internal.editors.result.test.TestProvider 

windows build had errors but they do not seem related to my changes.  new tests ran successfully.

**Review by jimblye — APPROVED** [review]
Approved, though I don't understand why the customer wanted a relative path.   They could always make the path relative themselves.   I mean, no one is stopping them from specifying the server working dir relative to a known directory.  I mean they could specify something like this:   ${SERVER_OUTPUT_DIR}/jvm_dir

But now, since we've forced it to be relative to the output directory, they don't have the option of putting it outside the output directory, unless they do something hokey with ../../..

Although, I don't know if anyone will need the flexibility.

**Comment by hlhoots** [comment]
#build

File: dev/com.ibm.ws.kernel.boot.ws-server/publish/bin/server [hunk 1]
```diff
@@ -808,12 +815,19 @@ readNativeFile() {
...
+## serverWorkingDirectory: Change the JVM working directory to ${SERVER_WORKING_DIR}
 serverWorkingDirectory()
 {
-  # Change the working directory to ${SERVER_OUTPUT_DIR}.
-  mkdirs "${SERVER_OUTPUT_DIR}"
-  cd "${SERVER_OUTPUT_DIR}"
+  if [[ -z "${SERVER_WORKING_DIR}" ]]; then
+    SERVER_WORKING_DIR=$SERVER_OUTPUT_DIR
+  else
+    if [[ "${SERVER_WORKING_DIR:0:1}" != "/" && "${SERVER_WORKING_DIR:0:1}" != ~[/a-z] ]]; then
```
Comment (covener) [inline dev/com.ibm.ws.kernel.boot.ws-server/publish/bin/server]: this (the slicing) doesn't seem portable

**Comment by hlhoots** [comment]
#build

**Comment by hlhoots** [comment]
https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_dXIx0EYyEeyda8UYcX_yfQ

**Comment by hlhoots** [comment]
#build

**Comment by hlhoots** [comment]
windows build:  https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_k7M-IEcMEeyda8UYcX_yfQ

**Comment by hlhoots** [comment]
build issues.

#build

**Comment by hlhoots** [comment]
windows: https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_zD2r4EexEeyda8UYcX_yfQ

**Comment by hlhoots** [comment]
#build

**Comment by hlhoots** [comment]
windows build: https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=__noKEEikEeyda8UYcX_yfQ

**Comment by hlhoots** [comment]
Good linux build.  Windows build had known errors, but new tests all passed.  Should be good to merge at this point.

**Review by cbridgha — APPROVED** [review]
Looks good

**Comment by hlhoots** [comment]
#run-libby-bot

**Pull request merged**